tests/MongoDB.Driver.Tests/ReadPreferenceOnStandaloneTests.cs

```diff
@@ -0,0 +1,86 @@
+/* Copyright 2019-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using FluentAssertions;
+using MongoDB.Bson;
+using MongoDB.Bson.TestHelpers.XunitExtensions;
+using MongoDB.Driver.Core;
+using MongoDB.Driver.Core.Clusters;
+using MongoDB.Driver.Core.Events;
+using MongoDB.Driver.Core.Misc;
+using MongoDB.Driver.TestHelpers;
+using Xunit;
+
+namespace MongoDB.Driver.Tests
+{
+    public class ReadPreferenceOnStandaloneTests
+    {
+        [Theory]
+        [ParameterAttributeData]
+        public void ReadPreference_should_not_be_sent_to_standalone_server(
+            [Values(false, true)] bool async)
+        {
+            var eventCapturer = new EventCapturer().Capture<CommandStartedEvent>(e => e.CommandName.Equals("find") || e.CommandName.Equals("$query"));
+            using (var client = CreateDisposableClient(eventCapturer, ReadPreference.PrimaryPreferred))
+            {
+                var database = client.GetDatabase(DriverTestConfiguration.DatabaseNamespace.DatabaseName);
+                var collection = database.GetCollection<BsonDocument>(DriverTestConfiguration.CollectionNamespace.CollectionName);
+
+                if (async)
+                {
+                    var _ = collection.FindAsync("{ x : 2 }").GetAwaiter().GetResult();
+                }
+                else
+                {
+                    var _ = collection.FindSync("{ x : 2 }");
+                }
+
+                BsonDocument sentCommand = ((CommandStartedEvent)eventCapturer.Events[0]).Command;;
```

Comment:
Also, maybe we can have only a `CommandStartEvent` object in this line and then use this event object to get `Command` and `CommandName` in the lines below. It will avoid us to have typecasting in the lines `60` and `64`  

Or we can create a new variable `sentCommandName` here.

Reply:
Done, that's a lot cleaner - thanks!